src/main/java/com/flansmod/common/driveables/EntityDriveable.java

```diff
@@ -857,6 +772,46 @@ public Vector3f getLookVector(DriveablePosition dp)
 		return axes.getXAxis();
 	}
 	
+	private void shootProjectile(final Integer slot, Vector3f gunVec, Vector3f lookVector, DriveableType type, Boolean secondary, float speed)
+	{
+		ItemStack bullet = driveableData.getStackInSlot(slot);
+		ItemBullet bulletItem = (ItemBullet)bullet.getItem();
+		int damageMultiplier = secondary ? type.damageModifierSecondary : type.damageModifierPrimary;
+
+		FireableGun fireableGun = new FireableGun(bulletItem.type, bulletItem.type.damageVsLiving*damageMultiplier, bulletItem.type.damageVsDriveable*damageMultiplier, bulletItem.type.bulletSpread, speed);
+		//TODO unchecked cast, grenades wont work (currently no vehicle with this feature exists)
+		FiredShot shot = new FiredShot(fireableGun, (BulletType) bulletItem.type, this, (EntityPlayerMP) getDriver());
+		
+		ShootBulletHandler handler = (Boolean isExtraBullet) ->
```

Comment:
Could this name be more descriptive? ShotFiredSuccessHandler or something? It doesn't say which bit of shooting is being handled

Reply:
Well it is just a handler called every time when a bullet is fired, so ShotFiredSuccessHandler doesn't really make sense to me. I have added a description in the interface, this may help understanding the function of the interface